crates/rattler_conda_types/src/conda_lock/content_hash.rs

```diff
@@ -0,0 +1,132 @@
+    #[error("the data for key `{0}` is required but missing")]
+    RequiredAttributeMissing(String),
+    #[error(transparent)]
+    JsonDecodeError(#[from] serde_json::Error),
+}
+
+/// This function tries to replicate the creation of the content-hashes
+/// like conda-lock does https://github.com/conda/conda-lock/blob/83117cb8da89d011a25f643f953822d5c098b246/conda_lock/models/lock_spec.py#L60
+/// so we need to recreate some python data-structures and serialize these to json
+pub fn calculate_content_data(
+    _platform: &Platform,
+    input_specs: &[MatchSpec],
+    channels: &[Channel],
+) -> Result<String, CalculateContentHashError> {
+    /// Selector taken from the conda-lock python source code
+    /// which we will just keep empty for now
+    #[derive(Serialize, Default, Debug)]
+    struct Selector {
+        platform: Option<Vec<String>>,
+    }
+
+    /// This is the equivalent of an VersionedDependency from
+    /// the conda-lock python source code
+    /// conda
+    #[derive(Serialize, Debug)]
+    struct CondaLockVersionedDependency {
+        build: Option<String>,
+        category: String,
+        conda_channel: Option<String>,
+        extras: Vec<String>,
+        manager: String,
+        name: String,
+        optional: bool,
+        selectors: Selector,
+        version: String,
+    }
+
+    /// Data for which the ContentHash hash has to be constructed
+    /// In python this is just a dictionary
+    #[derive(Serialize, Debug)]
+    struct ContentHashData {
+        channels: Vec<Channel>,
+        specs: Vec<CondaLockVersionedDependency>,
+    }
+
+    // Map our stuff to conda-lock types
+    let specs = input_specs
+        .iter()
+        .map(|spec| {
+            Ok(CondaLockVersionedDependency {
+                name: spec.name.clone().ok_or(
+                    CalculateContentHashError::RequiredAttributeMissing("name".to_string()),
+                )?,
+                manager: "conda".to_string(),
+                optional: false,
+                category: "main".to_string(),
+                extras: Default::default(),
+                selectors: Default::default(),
+                version: spec.version.as_ref().map(|v| v.to_string()).ok_or(
+                    CalculateContentHashError::RequiredAttributeMissing("version".to_string()),
+                )?,
+                build: spec.build.clone(),
+                conda_channel: None,
+            })
+        })
+        .collect::<Result<Vec<_>, CalculateContentHashError>>()?;
+
+    // In the python code they are also adding a virtual package hash
+    // For virtual packages overwritten by the user, we are skipping
+    // this for now
+    // TODO: Add default list of virtual packages and then create the content hashing
+
+    // Create the python dict
+    let content_hash_data = ContentHashData {
+        channels: channels.to_vec(),
+        specs,
+    };
+
+    // Create the json
+    Ok(serde_json::to_string(&content_hash_data)?
```

Comment:
Can we use the PrettyFormatter? https://docs.rs/serde_json/latest/serde_json/ser/struct.PrettyFormatter.html

Replacing `:` and `,` looks "dangerous" :D

Reply:
Let's see!

Reply:
Seems to work @baszalmstra @wolfv! Check it out :)